l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py

```diff
@@ -60,214 +55,234 @@
+                ('key2', '=', 'client_print_multi')]
 
-    def saveAttachment(self, cr, uid, context=None):
-        if context is None:
-            context = {}
+    report_print_menu = fields.Many2one(
+        comodel_name='ir.values',
+        domain=_domain_ir_values,
+        help='This report will be automatically included in the created XML')
 
-        number = self.number
+    def saveAttachment(self, fatturapa, number):
 
-        user_obj = self.pool['res.users']
-        company = user_obj.browse(cr, uid, uid).company_id
+        company = self.env.user.company_id
 
         if not company.vat:
-            raise orm.except_orm(
-                _('Error!'), _('Company TIN not set.'))
-        attach_obj = self.pool['fatturapa.attachment.out']
+            raise UserError(
+                _('Company %s TIN not set.') % company.name)
+        if (
+                    company.fatturapa_sender_partner and not
+                company.fatturapa_sender_partner.vat
+        ):
+            raise UserError(
+                _('Partner %s TIN not set.')
+                % company.fatturapa_sender_partner.name
+            )
+        vat = company.vat
+        if company.fatturapa_sender_partner:
+            vat = company.fatturapa_sender_partner.vat
+        vat = vat.replace(' ', '').replace('.', '').replace('-', '')
+        attach_obj = self.env['fatturapa.attachment.out']
         attach_vals = {
-            'name': '%s_%s.xml' % (company.vat, str(number)),
-            'datas_fname': '%s_%s.xml' % (company.vat, str(number)),
-            'datas': base64.encodestring(self.fatturapa.toxml("latin1")),
+            'name': '%s_%s.xml' % (vat, str(number)),
+            'datas_fname': '%s_%s.xml' % (vat, str(number)),
+            'datas': base64.encodestring(fatturapa.toxml("UTF-8")),
         }
-        attach_id = attach_obj.create(cr, uid, attach_vals, context=context)
-
-        return attach_id
+        return attach_obj.create(attach_vals)
 
-    def setProgressivoInvio(self, cr, uid, context=None):
-        if context is None:
-            context = {}
+    def setProgressivoInvio(self, fatturapa):
 
-        user_obj = self.pool['res.users']
-        company = user_obj.browse(cr, uid, uid).company_id
-        sequence_obj = self.pool['ir.sequence']
+        company = self.env.user.company_id
         fatturapa_sequence = company.fatturapa_sequence_id
         if not fatturapa_sequence:
-            raise orm.except_orm(
-                _('Error!'), _('FatturaPA sequence not configured.'))
-        self.number = number = sequence_obj.next_by_id(
-            cr, uid, fatturapa_sequence.id, context=context)
-        self.fatturapa.FatturaElettronicaHeader.DatiTrasmissione.\
-            ProgressivoInvio = number
-        return True
+            raise UserError(
+                _('FatturaPA sequence not configured.'))
+        number = fatturapa_sequence.next_by_id(fatturapa_sequence.id)
```

Comment:
`fatturapa_sequence.id` non serve (è già il record stesso)

Reply:
In v8 serve passargli fatturapa_sequence altrimenti la funzione va in errore

Reply:
hai ragione